core/src/test/java/org/apache/calcite/test/TopDownOptTest.java

```diff
@@ -287,6 +289,119 @@
         .removeRule(EnumerableRules.ENUMERABLE_JOIN_RULE)
         .check();
   }
+
+  // Not push down sort for hash join in inner join case.
+  @Test void testHashJoinInnerJoinNotPushDownSort() {
```

Comment:
In fact, this test case is not valid (sort not pushdown due to cost-baed optimization).

I will remove this one.